reference-implementation/to-upstream-wpts/writable-streams/aborting.js

```diff
@@ -530,6 +530,76 @@ promise_test(t => {
 }, 'writer.ready should reject on controller error without waiting for underlying write');
 
 promise_test(t => {
+  let rejectWrite;
+  const ws = new WritableStream({
+    write() {
+      return new Promise((resolve, reject) => {
+        rejectWrite = reject;
+      });
+    }
+  });
+
+  let writePromise;
+  let writePromiseRejected = false;
+
+  let abortPromise;
+  let abortPromiseRejected = false;
+
+  let closedRejected = false;
+
+  const writer = ws.getWriter();
+
+  // Wait for ws to start
+  return flushAsyncEvents().then(() => {
+    writePromise = writer.write('a');
+    writePromise.catch(() => {
+      writePromiseRejected = true;
+    });
+
+    abortPromise = writer.abort(error1);
+    abortPromise.catch(() => {
+      abortPromiseRejected = true;
+    });
+
+    const writePromise2 = writer.write('a');
+
+    writer.closed.catch(() => {
+      closedRejected = true;
+    });
+
+    return Promise.all([
+      promise_rejects(t, new TypeError(), writePromise2, 'writePromise2 must be rejected'),
+      promise_rejects(t, new TypeError(), writer.ready, 'writer.ready must be rejected after abort()'),
+      flushAsyncEvents()
+    ]);
+  }).then(() => {
+    assert_false(writePromiseRejected, 'writePromise must not be rejected after abort()');
+    assert_false(closedRejected, 'closed must not be rejected after abort()');
+    assert_false(abortPromiseRejected, 'abortPromise must not be rejected after abort()');
+
+    rejectWrite(error2);
+
+    return flushAsyncEvents();
+  }).then(() => {
+    const writePromise3 = writer.write('a');
+
+    return Promise.all([
+      promise_rejects(t, new TypeError(), writePromise3, 'writePromise3 must be rejected'),
+      promise_rejects(t, new TypeError(), writer.ready, 'writer.ready must be rejected after write completion'),
+      promise_rejects(t, error2, writer.closed, 'writer.closed must be rejected after write completion'),
+      promise_rejects(t, error2, abortPromise, 'abortPromise must be rejected after write completion')
+    ]);
+  }).then(() => {
+    writer.releaseLock();
+
+    return Promise.all([
+      promise_rejects(t, new TypeError(), writer.ready, 'writer.ready must be reset to be rejected with a TypeError'),
```

Comment:
Nitpick:  the assertion description implies that it verifies more than it actually does. If we want to actually check that it's been reset, we have to save the previous value and verify it is not equal.